### PR TITLE
KTOR-556 Add Content-Length header for NoContent

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-556](https://youtrack.jetbrains.com/issue/KTOR-556)

### CODE REVIEW

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt [hunk 1]
```diff
@@ -90,4 +93,15 @@ class HeadersTest : ClientLoader() {
...
     }
+
+    @Test
+    fun testRequestHasContentLength() = clientTests {
+        test { client ->
+            val first = client.put<String>("$TEST_SERVER/headers")
+            assertEquals("0", first)
+
+            val second = client.post<String>("$TEST_SERVER/headers")
+            assertEquals("0", second)
```
Comment (cy6erGn0m) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt:104]: Can't we test GET and HEAD here as well to ensure that we don't generate `Content-Length` for these methods?

**Review by cy6erGn0m — CHANGES_REQUESTED** [review]

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt:104]: Sure, will add

File: ktor-client/ktor-client-android/jvm/src/io/ktor/client/engine/android/AndroidClientEngine.kt [hunk 2]
```diff
@@ -88,7 +93,7 @@ public class AndroidClientEngine(override val config: AndroidEngineConfig) : Htt
 
             val content: ByteReadChannel = connection.content(callContext, data)
             val headerFields: Map<String, List<String>> = connection.headerFields
-                .mapKeys { it.key?.toLowerCase() ?: "" }
+                .mapKeys { it.key?.lowercase(Locale.getDefault()) ?: "" }
```
Comment (rsinukov) [inline ktor-client/ktor-client-android/jvm/src/io/ktor/client/engine/android/AndroidClientEngine.kt:96]: Is there overload that does not need locale?

File: ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/tests/Headers.kt [hunk 3]
```diff
@@ -17,7 +18,16 @@ internal fun Application.headersTestServer() {
                 call.response.header("X-Header-Single-Value", "foo")
                 call.response.header("X-Header-Double-Value", "foo")
                 call.response.header("X-Header-Double-Value", "bar")
-                call.respond(HttpStatusCode.OK, "OK")
+                val message = call.request.header(HttpHeaders.ContentLength) ?: ""
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/tests/Headers.kt:21]: Minor: maybe create a special route for `content-length` header and not modify this one? But up to you.

**Review by rsinukov — APPROVED** [review]

File: ktor-client/ktor-client-android/jvm/src/io/ktor/client/engine/android/AndroidClientEngine.kt (hunk 2 above)
Comment (e5l) [reply ktor-client/ktor-client-android/jvm/src/io/ktor/client/engine/android/AndroidClientEngine.kt:96]: no, it was deprecated

**Review by rsinukov — APPROVED** [review]

**Pull request merged**